src/snapred/backend/recipe/algorithm/LiteDataCreationAlgo.py

```diff
@@ -0,0 +1,90 @@
+import json
+
+from mantid.api import *
+from mantid.api import AlgorithmFactory, PythonAlgorithm, mtd
+from mantid.Kernel import Direction
+
+from snapred.backend.recipe.algorithm.MantidSnapper import MantidSnapper
+
+name = "LiteDataCreationAlgo"
+
+
+class LiteDataCreationAlgo(PythonAlgorithm):
+    def PyInit(self):
+        self.declareProperty("InputWorkspace", defaultValue="", direction=Direction.Input)
+        self.declareProperty("OutputWorkspace", defaultValue="", direction=Direction.Output)
+        self.setRethrows(True)
+        self.mantidSnapper = MantidSnapper(self, name)
+
+    def PyExec(self):
+        self.log().notice("Lite Data Creation START!")
+        # define super pixel dimensions
+        superPixel = [8, 8]
+
+        # get output workspace name
+        outputWorkspaceName = self.getProperty("OutputWorkspace").value
+
+        # load input workspace
+        inputWorkspace = self.getProperty("InputWorkspace").value
+        ws = self.mantidSnapper.mtd[inputWorkspace]
+
+        # clone the workspace
+        self.mantidSnapper.CloneWorkspace(
+            "Cloning input workspace for lite data creation...",
+            InputWorkspace=inputWorkspace,
+            OutputWorkspace=outputWorkspaceName,
+        )
+        self.mantidSnapper.executeQueue()
+        outputWorkspace = self.mantidSnapper.mtd(outputWorkspaceName)
+
+        # IEventWorkspace ID is equivalent to spec ID
+        numSpec = outputWorkspace.getNumberHistograms()
+
+        # iterate over IEventWorkspace IDs
+        for spec in range(numSpec):
+            # wsID = outputWorkspace.getSpectrum(spec)
+
+            # get event list per workspace
+            eventList = outputWorkspace.getEventList(workspace_index=spec)
+
+            # allocate array of event IDs
+            for event in eventList:
+                eventID = eventList.getDetectorIDs()
+                superID = self.getSuperID(nativeID=eventID, xdim=superPixel[0], ydim=superPixel[0])
+
+                # clear detector IDs and then add the new modified super ID
+                event.clearDetectorIDs()
+                event.addDetectorID(superID)
+
+        self.mantidSnapper.DeleteWorkspace("Cleaning up input workspace...", Workspace=ws)
+        self.mantidSnapper.executeQueue()
+
+        self.setProperty("OutputWorkspace", outputWorkspaceName)
```

Comment:
you may need to update the associated instrument stored on the output workspace, otherwise it may still think its the full SNAP instrument as opposed to the lite version.

Reply:
otherwise I think everything else is shaping up pretty good.

Reply:
you should just be able to use
 self.mantidSnapper.[LoadInstrument](https://docs.mantidproject.org/nightly/algorithms/LoadInstrument-v1.html)("Loading instrument...",, Workspace=outputWorkspace, Filename="/SNS/SNAP/shared/Calibration/Powder/SNAPLite.xml", RewriteSpectraMap=False)

Reply:
Should be added. Thank you!